src/uct/rocm/base/rocm_base.c

```diff
@@ -292,9 +292,44 @@ int uct_rocm_base_is_dmabuf_supported()
     int dmabuf_supported = 0;
 
 #if HAVE_HSA_AMD_PORTABLE_EXPORT_DMABUF
-    dmabuf_supported = 1;
+    const char kernel_opt1[28] = "CONFIG_DMABUF_MOVE_NOTIFY=y";
+    const char kernel_opt2[20] = "CONFIG_PCI_P2PDMA=y";
+    int found_opt1             = 0;
+    int found_opt2             = 0;
+    FILE *fp;
+    struct utsname utsname;
+    char kernel_conf_file[128];
+    char buf[256];
+
+    if (uname(&utsname) == -1) {
+        ucs_trace("could not get kernel name");
+        goto exit;
+    }
+
+    ucs_snprintf_safe(kernel_conf_file, sizeof(kernel_conf_file),
+                      "/boot/config-%s", utsname.release);
+    errno = 0;
```

Comment:
why needed?

Reply:
Some programming guides (e.g. https://www.ibm.com/docs/en/rdfi/9.6.0?topic=errors-checking-errno-value) recommend resetting errno to zero before calling a function that uses errno.

Reply:
This is if errno is used as indication if error happened or not.
In case we already know the function failed - then errno is set to the error value - see https://man7.org/linux/man-pages/man3/fopen.3.html section "ERRORS" for example

Reply:
@yosefe ok, sounds good, I removed the line.